[license_worker] KeyError: 'getpwuid(): uid not found: 1000090000'

This seems to be something dsaas/openshift specific as I cant reproduce it locally. 

```
2017-06-21 15:20:20,691 [INFO] celery.app.trace: Task selinon.Dispatcher[32e235bf-b66a-4987-b701-0fa131b3e2c4] retry: Retry in 2s
2017-06-21 15:20:21,733 [INFO] celery.worker.strategy: Received task: selinon.SelinonTaskEnvelope[2239c5e0-1eae-4c26-89d5-207205ffaf07]  
{"task_name": "source_licenses", "task_id": "2239c5e0-1eae-4c26-89d5-207205ffaf07", "parent": {}, "node_args": {"version": "1.5.2.RELEASE", "force": true, "document_id": 755970, "_audit": {"version": "v1", "ended_at": "2017-06-21T15:20:12.037380", "started_at": "2017-06-21T15:20:08.986963"}, "_release": "maven:org.springframework.boot:spring-boot-test-autoconfigure:1.5.2.RELEASE", "name": "org.springframework.boot:spring-boot-test-autoconfigure", "ecosystem": "maven"}, "event": "TASK_START", "flow_name": "bayesianAnalysisFlow", "queue": "prod_ingestion_LicenseCheckTask_v0", "dispatcher_id": "f42514a8-36eb-4d81-a274-37520d0f7314"}
Traceback (most recent call last):
  File "/opt/scancode-toolkit/bin/scancode", line 11, in <module>
    load_entry_point('scancode-toolkit', 'console_scripts', 'scancode')()
  File "/opt/scancode-toolkit/lib/python2.7/site-packages/pkg_resources/__init__.py", line 561, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/opt/scancode-toolkit/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2649, in load_entry_point
    return ep.load()
  File "/opt/scancode-toolkit/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2303, in load
    return self.resolve()
  File "/opt/scancode-toolkit/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2309, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/opt/scancode-toolkit/src/scancode/__init__.py", line 35, in <module>
    from commoncode import fileutils
  File "/opt/scancode-toolkit/src/commoncode/fileutils.py", line 49, in <module>
    from commoncode import system
  File "/opt/scancode-toolkit/src/commoncode/system.py", line 121, in <module>
    username = getpass.getuser()
  File "/usr/lib64/python2.7/getpass.py", line 158, in getuser
    return pwd.getpwuid(os.getuid())[0]
KeyError: 'getpwuid(): uid not found: 1000090000'
2017-06-21 15:20:23,315 [WARNING] cucoslib.utils: command ['/opt/scancode-toolkit/scancode', '--license', '--license-score', '20', '--only-findings', '--processes', '1', '--quiet', '--strip-root', '--timeout', '10', '/var/lib/cucos-lib/worker_data/maven/org.springframework.boot:spring-boot-test-autoconfigure/1.5.2.RELEASE/extracted_jar'] ended with 1

2017-06-21 15:20:23,316 [ERROR] cucoslib.utils: exception is fatal
2017-06-21 15:20:23,316 [ERROR] LicenseCheckTask: License scan failed
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/cucoslib/utils.py", line 237, in get_command_output
    out = check_output(args, universal_newlines=True, **kwargs)
  File "/usr/lib64/python3.4/subprocess.py", line 617, in check_output
    raise CalledProcessError(retcode, process.args, output=output)
subprocess.CalledProcessError: Command '['/opt/scancode-toolkit/scancode', '--license', '--license-score', '20', '--only-findings', '--processes', '1', '--quiet', '--strip-root', '--timeout', '10', '/var/lib/cucos-lib/worker_data/maven/org.springframework.boot:spring-boot-test-autoconfigure/1.5.2.RELEASE/extracted_jar']' returned non-zero exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/cucoslib/workers/license.py", line 89, in execute
    timeout=600)
  File "/usr/lib/python3.4/site-packages/cucoslib/utils.py", line 221, in get_command_output
    return get_command_output(args, graceful, is_json, **kwargs)
  File "/usr/lib/python3.4/site-packages/cucoslib/utils.py", line 247, in get_command_output
    raise TaskError("Error during running command %s: %r" % (args, ex.output))
cucoslib.errors.TaskError: Error during running command ['/opt/scancode-toolkit/scancode', '--license', '--license-score', '20', '--only-findings', '--processes', '1', '--quiet', '--strip-root', '--timeout', '10', '/var/lib/cucos-lib/worker_data/maven/org.springframework.boot:spring-boot-test-autoconfigure/1.5.2.RELEASE/extracted_jar']: ''
```

--- a/cucoslib/utils.py
+++ b/cucoslib/utils.py
@@ -1,3 +1,4 @@
+import getpass
 import json
 import logging
 import datetime
@@ -143,6 +144,24 @@ def tempdir():
         yield dirpath
     finally:
         shutil.rmtree(dirpath)
+
+
+@contextmanager
+def username():
+    """ workaround for failing getpass.getuser()
+        http://blog.dscpl.com.au/2015/12/unknown-user-when-running-docker.html
+    """
+    user = ''
+    try:
+        user = getpass.getuser()
+    except KeyError:
+        os_environ['LOGNAME'] = 'f8aworker'
+
+    try:
+        yield
+    finally:
+        if not user:
+            del os_environ['LOGNAME']
 
 
 def assert_not_none(name, value):

--- a/cucoslib/workers/license.py
+++ b/cucoslib/workers/license.py
@@ -3,7 +3,7 @@ Uses ScanCode toolkit to detect licences in source code.
 """
 
 import os
-from cucoslib.utils import TimedCommand
+from cucoslib.utils import TimedCommand, username
 from cucoslib.base import BaseTask
 from cucoslib.schemas import SchemaRef
 from cucoslib.object_cache import ObjectCache
@@ -86,10 +86,11 @@ class LicenseCheckTask(BaseTask):
                        # Stop scanning a file if scanning takes longer than a timeout in seconds
                        '--timeout', SCANCODE_TIMEOUT,
                        cache_path]
-            output = TimedCommand.get_command_output(command,
-                                                     graceful=False,
-                                                     is_json=True,
-                                                     timeout=1200)
+            with username():
+                output = TimedCommand.get_command_output(command,
+                                                         graceful=False,
+                                                         is_json=True,
+                                                         timeout=1200)
             details = self.process_output(output)
             result_data['details'] = details
             result_data['status'] = 'success'

